model/src/main/java/org/projectnessie/model/IcebergTable.java

```diff
@@ -56,24 +56,34 @@
 @JsonTypeName("ICEBERG_TABLE")
 public abstract class IcebergTable extends IcebergContent {
 
+  @Override
+  @Nullable
+  @Value.Parameter(order = 1)
+  public abstract String getId();
+
   /**
    * Location where Iceberg stored its {@code TableMetadata} file. The location depends on the
    * (implementation of) Iceberg's {@code FileIO} configured for the particular Iceberg table.
    */
   @NotNull
   @NotBlank
+  @Value.Parameter(order = 2)
   public abstract String getMetadataLocation();
 
   /** Corresponds to Iceberg's {@code currentSnapshotId}. */
+  @Value.Parameter(order = 3)
   public abstract long getSnapshotId();
 
   /** Corresponds to Iceberg's {@code currentSchemaId}. */
+  @Value.Parameter(order = 4)
   public abstract int getSchemaId();
 
   /** Corresponds to Iceberg's {@code defaultSpecId}. */
+  @Value.Parameter(order = 5)
   public abstract int getSpecId();
 
   /** Corresponds to Iceberg's {@code defaultSortOrderId}. */
+  @Value.Parameter(order = 6)
   public abstract int getSortOrderId();
```

Comment:
How will generated `of` method behave when parameters are added/removed? I guess that would be addition risk of confusion in API classes since `javac` will only validate the total number of args and that the types match.